elifetools/tests/test_xmlio.py

```diff
@@ -109,5 +109,58 @@ def test_append_minidom_xml_to_elementtree_xml(self):
         rough_string = ElementTree.tostring(parent).decode('utf-8')
         self.assertEqual(rough_string, '<root><i>Text</i> and tail.<span><i>and</i> some <i>XML</i>.</span><span class="span"> <i>more</i> text</span></root>')
 
+    @unpack
+    @data(
+        # example of removing and adding the same heading tags
+        (u'<article><front><article-meta><article-categories><subj-group subj-group-type="display-channel"><subject>Research Article</subject></subj-group><subj-group subj-group-type="heading"><subject>Cell Biology</subject></subj-group><subj-group subj-group-type="heading"><subject>Plant Biology</subject></subj-group></article-categories></article-meta></front></article>',
```

Comment:
Is this first argument in `@data(..)` the whole article XML? Are they all different in all data tags? Perhaps it would make it more readable if it was reading from a file instead? A variable that looks at a file?

Reply:
Each one is a snippet of XML, in this case it has lots of parent tags because the parser is (I think) accustomed to find these subject tags within the ``<article-meta>``.

These large strings of content in the test cases are ugly, and they could be done as separate files, except managing each as a separate file and having to refer to them and the expected output, I'm not sure what is the best way to represent these.

Maybe more opinions on whether strings or files are better as input here? I do not feel strongly one way or the other.

Reply:
I agree it would be cleaner if it was just referring fixture names, either being actual fixtures (as in pytest) or file names to be parsed. It adds some management overhead, but it would make the tests clearer I think.

Anything up to a few lines I would put in a fixture and refer to the variable, anything longer I would normally dump in a file.